inventory.py

```diff
@@ -69,6 +76,79 @@ def slim_down_course_data(course_data: Sequence[Dict]) -> Sequence[Dict]:
         slim_course_dicts.append(slim_course_dict)
     return slim_course_dicts
 
+def get_next_page_url(response):
+    """
+    get the next page url from the Http response headers
+    :param response:
+    :type response: requests
+    :return: next_page_url
+    :rtype: str
+    """
+    logging.debug(get_next_page_url.__name__ + '() called')
+
+    if not response.links:
+        logging.debug('The api call do not have Link headers')
+        return None
+
+    for page in response.links:
+        if 'next' in page:
+            return response.links['next']['url']
+
+
+def handle_request_if_failed(response):
+    if response.status_code != 200:
+        return False
+    else:
+        return True
+
+
+def get_course_publish_unpublish_date(df_row: str, publish_or_unpublish_date, next_page_url = None)->str:
+    logger.debug(df_row)
+    if df_row['course_workflow_state'] == 'available':
+        event_type_to_check_for = 'published'
+    if df_row['course_workflow_state'] == 'unpublished':
+        event_type_to_check_for = 'claimed'
+
+    course_id= df_row['course_id']
+
+    if next_page_url is not None:
+        url = next_page_url
+    else:
+        url = f"{API_SCOPE_PREFIX}/audit/course/courses/{course_id}?per_page={PER_PAGE_COUNT}"
+
+    try:
+        response = make_audit_logs_api_request(url)
+
+    except (Exception) as e:
+        logger.exception('getting sections has erroneous response ' + e.message)
+        return publish_or_unpublish_date
```

Comment:
What exception are you seeing here? Any chance it's the `JSONDecodeError` I'm catching in the `make_request_using_api_utils` function?

Reply:
This is the exception return from the API utils and it looks like it is a generic exception.

Reply:
What is a generic exception?

Reply:
https://github.com/tl-its-umich-edu/api-utils-python/blob/master/umich_api/api_utils.py#L132